tools/csvstoskyui/csvstoskyui.py

```diff
@@ -0,0 +1,143 @@
+import csv
+import json
+import re
+import argparse
+
+"""
+The content of topologyrules.conf file includes a python dictionary with the following keys: 
+
+filename    - The name of the file to read the csv information from
+              example: "netbox/netbox_sites.csv"
+show        - Boolean that enables or disable usage of this csv source
+              example: False --> will result in skipping this csv source  
+valuesmappingcopy - dictionary of csv values to copy from csv to json
+              example: {"group_name": "Name"} --> Copy csv values from column 'group_name' to be called 'Name' in json 
+valuesfilter - dictionary of csv values to filter by using regex
+              example: "site": ".*ams01" --> Using only csv entries with site equal to regex   
+edges        - Create edges between nodes (list of edges)
+              example: {"from":["site"], "to": {"site": ["name"]}, "RelationType": "ownership"} --> Will create
+              an edge from every entry in csv having 'site' field that is equal to field 'name' in the csv source
+              'site' and set RelationType of the edge to be ownership (Note: can be list of fields to compare as needed)
+"""
+
+## build nodes
+def buildNodes(netboxtopologyfiles):
+    jsonnodes = []
+    for key in netboxtopologyfiles:
+        jsontopology = []
+        if not ("show" in netboxtopologyfiles[key]) or not (netboxtopologyfiles[key]["show"]):
+            continue
+
+        csv_reader = csv.DictReader(open(netboxtopologyfiles[key]["filename"]))
+        for idx, raw in enumerate(csv_reader):
+            jsondump = {"Metadata": raw,
+                        "ID": (key + "-" + str(idx)).encode("hex"),
+                        "Origin": "SoftLayer"}
+            jsondump["Metadata"].update({"Type": key})
+            if "valuesfilter" in netboxtopologyfiles[key]:
+                filter_value = False
+                for filterkey, filtervalue in netboxtopologyfiles[key]["valuesfilter"].iteritems():
+                    try:
+                        if not re.search(filtervalue, raw[filterkey]):
+                            filter_value = True
+                    except TypeError:
+                        filter_value = True
+                if filter_value:
+                    continue
+            if "valuesmappingcopy" in netboxtopologyfiles[key]:
+                for mapkey, mapvalue in netboxtopologyfiles[key]["valuesmappingcopy"].iteritems():
```

Comment:
you should check (or assert) that its' a map

Reply:
This is correct to many fields and sub-fields ... need a 'project' to check the scheme ... something like https://pypi.org/project/jsonschema/ maybe I will do that sometime in the future ... but I do not have the time at this point to do that. agree with the comment.